GitCommands/CustomDiffMergeToolCache.cs

```diff
@@ -48,6 +48,8 @@ public async Task<IEnumerable<string>> GetToolsAsync(GitModule module, int delay
                 return _tools;
             }
 
+            cancellationToken.ThrowIfCancellationRequested();
+
             // The command will compete with other resources, avoid delaying startup
             // Do not bother with cancel tokens
             await Task.Delay(delay, cancellationToken);
```

Comment:
Have you seen signs that the .NET threading function do not do their job?
All these `ThrowIfCancellationRequested` on the background thread, are added right before calls with cancellation.

I guess adding a few menu items on the main thread cannot cause hangs.
`SwitchToMainThreadAsync` should have checked the cancellation token before it yields.

So - as Gerhard pointed out - the only remaining thing from this PR is #9818 IMHO.